src/Types/Enums/EdgeColumn.php

```diff
@@ -5,20 +5,27 @@
 namespace Mrap\GraphCool\Types\Enums;
 
 use GraphQL\Type\Definition\EnumType;
+use Mrap\GraphCool\Definition\Model;
+use Mrap\GraphCool\Definition\Relation;
 use function Mrap\GraphCool\model;
 
-class EdgeColumnType extends EnumType
+class EdgeColumn extends EnumType
 {
 
-    public function __construct(string $name)
+    public function __construct(Relation $relation)
     {
-        $names = explode('__', substr($name, 1, -10), 2);
-        $parentModel = model($names[0]);
-
-        $relation = $parentModel->{$names[1]};
+        $config = [
+            'name' => '_' . $relation->namekey. 'EdgeColumn',
```

Comment:
ist eig schade das man nicht `"_{$relation->namekey}EdgeColumn"` schreiben kann

Reply:
Ich bevorzuge das so explizit zu schreiben.

So sehe ich zwei statische strings, die mittels des operators `.` mit einer echten variable verknüpft werden.

Anders ist es halt ein magischer string, der geparsed wird und falls drin im text was enthalten ist, was einen variablen-namen matched, wird dass dann mit dem variablen-inhalt string-replaced... das is halt nicht so meins ;)